programs/test/benchmark.c

```diff
@@ -1201,33 +1201,60 @@ int main(int argc, char *argv[])
             }
 
             mbedtls_ecdh_init(&ecdh_srv);
-            mbedtls_ecdh_init(&ecdh_cli);
+
+            CHECK_AND_CONTINUE(mbedtls_ecdh_setup(&ecdh_srv, curve_info->grp_id));
+            CHECK_AND_CONTINUE(mbedtls_ecdh_make_params(&ecdh_srv, &params_len, buf_srv,
+                                                        sizeof(buf_srv), myrand, NULL));
 
             mbedtls_snprintf(title, sizeof(title), "ECDHE-%s", curve_info->name);
             TIME_PUBLIC(title,
-                        "full handshake",
+                        "ephemeral handshake",
```

Comment:
Preexisting: I find the presentation of ECDH timings as “handshakes” weird. To me, the operations involved in a key agreement are:

1. generate the private key;
2. calculate the corresponding public key;
3. calculate the shared secret.

“Ephemeral handshake” = 1 + 2 + 3, “static handshake” = 3 alone. Ok, but that's a weird way to put it.

Reply:
Agreed. If you have a better name to suggest, please do. Otherwise I'm inclined to wait until we convert this to PSA crypto at which point I think we can just use the name of the function: `psa_generate_key()`, `psa_export_public_key()`, and `psa_raw_key_agreement()` for the 3 steps you mention, which I think will be the most precise least ambiguous way of presenting things.